pkg/authn/k8schain/k8schain.go

```diff
@@ -132,6 +121,23 @@ func NewNoClient(ctx context.Context) (authn.Keychain, error) {
 	return New(ctx, nil, Options{})
 }
 
+// NewFromPullSecrets returns a new authn.Keychain suitable for resolving image references as
+// scoped by the pull secrets.
+func NewFromPullSecrets(ctx context.Context, pullSecrets []corev1.Secret) (authn.Keychain, error) {
+	once.Do(func() {
+		keyring = credentialprovider.NewDockerKeyring()
+	})
+
+	// Extend the default keyring with the pull secrets.
+	kr, err := credentialprovidersecrets.MakeDockerKeyring(pullSecrets, keyring)
```

Comment:
You're okay with the default keyring here being the magical `NewDockerKeyring()` thing?

I'm starting to lean away from that (https://github.com/google/go-containerregistry/issues/1042) because it's caused problems (https://github.com/knative/serving/issues/10259).

Reply:
My goal for this PR was to not change the current behavior. I agree with the spirit of #1042